File: articles/herd-behavior-financial-markets/fortran/herd_buy_probability.f90
program herd_buy_probability
  implicit none
  real :: fundamental, private_signal, herd_signal, perceived_risk
  real :: alpha, beta, gamma, utility, probability

  fundamental = 0.15
  private_signal = 0.20
  herd_signal = 0.70
  perceived_risk = 0.10
  alpha = 1.0
  beta = 1.4
  gamma = 0.8

  utility = fundamental + alpha * private_signal + beta * herd_signal - gamma * perceived_risk
  probability = 1.0 / (1.0 + exp(-utility))

  print *, "Synthetic herd buy probability:", probability
end program herd_buy_probability

File: articles/herd-behavior-financial-markets/rust/herd_buy_probability.rs
fn logistic(x: f64) -> f64 {
    1.0 / (1.0 + (-x).exp())
}

fn buy_probability(
    fundamental: f64,
    private_signal: f64,
    herd_signal: f64,
    perceived_risk: f64,
    alpha: f64,
    beta: f64,
    gamma: f64,
) -> f64 {
    let utility = fundamental + alpha * private_signal + beta * herd_signal - gamma * perceived_risk;
    logistic(utility)
}

fn main() {
    println!(
        "Synthetic herd buy probability: {:.3}",
        buy_probability(0.15, 0.20, 0.70, 0.10, 1.0, 1.4, 0.8)
    );
}
